internal/pkg/deploy/pipeline.go

```diff
@@ -136,6 +144,40 @@ func (s *CodeCommitSource) parseRepo() (string, error) {
 	return matches["repo"], nil
 }
 
+// parseOwnerAndRepo parses the owner and repo name from the BB repo URL, which was formatted and assigned in cli/pipeline_init.go.
+func (s *BitbucketSource) parseOwnerAndRepo() (owner, repo string, err error) {
+	if s.RepositoryURL == "" {
+		return "", "", fmt.Errorf("unable to locate the repository")
+	}
+
+	match := bbRepoExp.FindStringSubmatch(s.RepositoryURL)
+	if len(match) == 0 {
+		return "", "", fmt.Errorf(fmtInvalidRepo, s.RepositoryURL)
+	}
+
+	matches := make(map[string]string)
+	for i, name := range bbRepoExp.SubexpNames() {
+		if i != 0 && name != "" {
+			matches[name] = match[i]
+		}
+	}
+	return matches["owner"], matches["repo"], nil
+}
+
+func (s *BitbucketSource) ConnectionName() (string, error) {
+	owner, repo, err := s.parseOwnerAndRepo()
+	if err != nil {
+		return "", fmt.Errorf("parse owner and repo to generate connection name: %w", err)
+	}
+	if len(owner) > 5 {
+		owner = owner[:5]
+	}
+	if len(repo) > 18 {
+		repo = repo[:18]
```

Comment:
Maybe consider giving 5 and 18 more human readable const name and a comment specifying the connection name length limit with the reference source.

Reply:
Ah, good ideas. Done.